### PR TITLE
Task- 5: Add Unit Testing for Budget Tracker 

### PR DESCRIPTION
**Pull request by ads5331145**
This task implements unit tests for the Personal Budget Tracking System using Python's unittest framework.

✅ Covered Test Cases:
1. Verify that income and expense transactions are stored correctly.
2. Ensure balance calculation is accurate for various transaction combinations.
3. Test handling of invalid inputs like negative amounts and non-numeric values.
4. Validate file operations such as saving to and loading from CSV files.
5. Confirm that printed outputs for transactions and warnings are correctly displayed.

All test logic is contained in the `test_budget_tracker.py` file, located inside the `task-5/` directory.

### CODE REVIEW

File: Task-5/test_budget_tracker.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            {"type": "income", "amount": 1000.0, "source": "Job", "category": ""},
+            {"type": "expense", "amount": 300.0, "source": "", "category": "expense"}
+        ]
+        m = mock_open()
+        with patch("builtins.open", m):
+            save_transactions_to_file(transactions, "test.csv")
+        m.assert_called_once_with("test.csv", mode="w", newline="")
+
+    def test_load_transactions_from_file(self):
+        data = "type,amount,source,category\nincome,1000.0,Job,\nexpense,300.0,,expense\n"
```
Comment (sf-ayushisri) [inline Task-5/test_budget_tracker.py:87]: Test edge case for malformed CSV row as well.

**Review by sf-ayushisri — CHANGES_REQUESTED** [review]

File: Task-5/test_budget_tracker.py (hunk 1 above)
Comment (ads5331145) [reply Task-5/test_budget_tracker.py:87]: Tested edge case for Added malformed CSV row
